File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/_4_BastardInjection/Inbound/UdpSocket.java
package com.github.grzesiek_galezowski.BastardInjection._4_BastardInjection.Inbound;

import com.github.grzesiek_galezowski.BastardInjection._4_BastardInjection.Core.IDisposable;

interface IInputSocket extends IDisposable //first error - do not implement disposabled in interfaces
  {
    boolean Receive(byte[] frameData);
  }

  class UdpSocket implements IInputSocket
  {
    public boolean Receive(byte[] frameData)
    {
      //todo generate random the java way new Random().NextBytes(frameData);
      return true;
    }

    public void Dispose()
    {
      System.out.println("Disposed");
    }
  }

File: CSharp/DependencyInjection/BastardInjection/_4_BastardInjection/Inbound/UdpSocket.cs
using System;

namespace BastardInjection._4_BastardInjection.Inbound
{
  internal interface IInputSocket : IDisposable //first error - do not implement disposabled in interfaces
  {
    bool Receive(out byte[] frameData);
  }

  class UdpSocket : IInputSocket
  {
    public bool Receive(out byte[] frameData)
    {
      frameData = new byte[100];
      new Random().NextBytes(frameData);
      return true;
    }

    public void Dispose()
    {
      Console.WriteLine("Disposed");
    }
  }
}